discovery-infra/download_logs.py

```diff
@@ -59,13 +59,34 @@ def min_number_of_log_files(cluster, is_controller_expected):
     else:
         return  len(cluster['hosts'])
 
-def download_logs(client: InventoryClient, cluster: dict, dest: str,
-                  must_gather: bool, retry_interval: int = RETRY_INTERVAL):
+def is_update_needed(output_folder: str, update_on_events_update: bool, client: InventoryClient, cluster: dict):
+
+    if not os.path.isdir(output_folder):
+        return True
+    if not update_on_events_update:
+        return False
+
+    destination_event_file_path = os.path.join(output_folder, f"cluster_{cluster['id']}_events.json")
```

Comment:
Make a format variable or a template, because the events file path is being processed twice.